### PR TITLE
Update eslint plugins

### PR DESCRIPTION
**Pull request by gregaubert**

### CODE REVIEW

File: components/controls/buttons.css [hunk 1]
```diff
@@ -185,6 +185,7 @@
 }
 
 /* #region .button-group */
+/* TODO drop usage of this class in SQ (already dropped from SC) */
```
Comment (philippe-perrin-sonarsource) [inline components/controls/buttons.css:188]: @jeremy-davis-sonarsource @mathieu-suen-sonarsource Have we decided to move forward on this ?

File: helpers/__tests__/request-test.ts [hunk 2]
```diff
@@ -95,7 +95,7 @@ describe('parseError', () => {
     ).resolves.toBe('Error1. Error2');
   });
 
-  it('should parse error and return default message', () => {
+  it('should parse error object and return default message', () => {
```
Comment (philippe-perrin-sonarsource) [inline helpers/__tests__/request-test.ts]: Not related to the PR: this test is incredibly difficult to understand IMO. Also, the loop only runs 2 times, so I'm not sure this really does test anything because the call after only is the third one. WDYT ?

File: helpers/__tests__/request-test.ts [hunk 3]
```diff
@@ -233,7 +233,8 @@ describe('requestTryAndRepeatUntil', () => {
       stopRepeat
     );
 
-    expect(promiseResult).rejects.toBe(undefined);
+    // eslint-disable-next-line jest/valid-expect
+    expect(promiseResult).rejects.toBeUndefined();
```
Comment (philippe-perrin-sonarsource) [inline helpers/__tests__/request-test.ts]: Removing this line make the next test fail ... :upside_down_face:

File: components/charts/AdvancedTimeline.tsx [hunk 4]
```diff
@@ -251,30 +251,33 @@ export default class AdvancedTimeline extends React.PureComponent<Props, State>
...
-            date.valueOf() - previousPoint.x.valueOf() <= nextPoint.x.valueOf() - date.valueOf())
-        ) {
-          idx--;
-        }
-        const selectedDate = firstSerie.data[idx].x;
-        const xPos = this.state.xScale(selectedDate);
-        this.setState({ selectedDate, selectedDateXPos: xPos, selectedDateIdx: idx });
-        if (updateTooltip) {
-          updateTooltip(selectedDate, xPos, idx);
+    this.setState(state => {
```
Comment (philippe-perrin-sonarsource) [inline components/charts/AdvancedTimeline.tsx:254]: NP: you probably could continue to avoid calling the `setState` if conditions (L256) are not fulfilled. The `return null` at the end of the `setState` is weird IMO.

**Review by philippe-perrin-sonarsource — COMMENTED** [review]
Thank you for taking the time to do implement those changes. I'm not giving you an approval because I think there is some issues with the `requests-test.ts` file. Happy to discuss it with you.
Also, you probably should ask for an SC's team member review.

File: helpers/__tests__/request-test.ts (hunk 2 above)
Comment (gregaubert) [reply helpers/__tests__/request-test.ts]: Yeah I agree it's not the best way of testing promises. I could replace everything with some async/await.

File: components/charts/AdvancedTimeline.tsx (hunk 4 above)
Comment (gregaubert) [reply components/charts/AdvancedTimeline.tsx:254]: The `return null` is used because we don't want to update the state, I think we could also return an empty object and it would do the same.

I can't move the full condition of (L256) outside the `setState` since it's based on the state. But ok I could split the condition to avoid calling the `setState` if the `firstSerie` is `undefined`, but then it add one more layer of `if` around everything and just make things a bit more complex for probably not much performance gain (just avoiding a function call basically)...

Comment (jeremy-davis-sonarsource) [reply components/charts/AdvancedTimeline.tsx:254]: Calling setState with null doesn't trigger a state change, so it's the basically the same as not calling it

File: helpers/__tests__/request-test.ts (hunk 2 above)
Comment (gregaubert) [reply helpers/__tests__/request-test.ts]: I'm not sure which loop you are talking about ? It's in a different test right ?

File: components/controls/buttons.css (hunk 1 above)
Comment (jeremy-davis-sonarsource) [reply components/controls/buttons.css:188]: We'll tackle that when we work on them (or containing page). So, no plan for a plain refactoring ticket.

**Review by jeremy-davis-sonarsource — APPROVED** [review]
LGTM

File: helpers/__tests__/request-test.ts (hunk 2 above)
Comment (philippe-perrin-sonarsource) [reply helpers/__tests__/request-test.ts]: The loop line 217. My understanding is that we want to leave room for more than 3 calls and check that only 3 calls happens. But I feel like we only leave room for 3 calls (2 in the loop, one after) and therefore we're not testing the edge case ('should stop after 3 calls').

Comment (gregaubert) [reply helpers/__tests__/request-test.ts]: Indeed, I fixed this by adding one more check at the end that it's not called anymore

**Review by philippe-perrin-sonarsource — APPROVED** [review]
LGTM
